CMakeLists.txt

```diff
@@ -46,65 +40,31 @@ endif()
 
 # this is needed for the #include directives with absolutes paths to work correctly; it must
 # also be set to PUBLIC since rellic-lift includes some files directly
-list(APPEND PROJECT_INCLUDEDIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/include)
+list(APPEND PROJECT_INCLUDEDIRECTORIES $<BUILD_INTERFACE:${RELLIC_SOURCE_DIR}/include>)
+list(APPEND PROJECT_INCLUDEDIRECTORIES $<INSTALL_INTERFACE:include>)
 
 #
 # libraries
 #
 
-# llvm
-find_package(LLVM REQUIRED CONFIG HINTS ${FINDPACKAGE_LLVM_HINTS})
-
-string(REPLACE "." ";" LLVM_VERSION_LIST ${LLVM_PACKAGE_VERSION})
-list(GET LLVM_VERSION_LIST 0 LLVM_MAJOR_VERSION)
-list(GET LLVM_VERSION_LIST 1 LLVM_MINOR_VERSION)
-
-set(LLVM_LIBRARIES
-  LLVMCore LLVMSupport LLVMAnalysis LLVMipo LLVMIRReader
-  LLVMBitReader LLVMBitWriter LLVMTransformUtils LLVMScalarOpts
-  LLVMLTO
-)
-
-list(APPEND PROJECT_LIBRARIES ${LLVM_LIBRARIES})
-list(APPEND PROJECT_DEFINITIONS ${LLVM_DEFINITIONS})
-list(APPEND PROJECT_INCLUDEDIRECTORIES ${LLVM_INCLUDE_DIRS})
-
-# clang
-find_package(Clang REQUIRED)
-set(CLANG_LIBS clangIndex clangCodeGen clangASTMatchers clangTooling)
-list(APPEND PROJECT_LIBRARIES ${CLANG_LIBS})
-
-# z3
-find_package(Z3 CONFIG 4.7.1 REQUIRED)
-list(APPEND PROJECT_LIBRARIES z3::libz3)
-
-# Potentially fixup bad LLVM z3 linkage
-get_target_property(LLVMSupport_LIBS LLVMSupport INTERFACE_LINK_LIBRARIES)
-list(FIND LLVMSupport_LIBS "Z3" _index)
-if (${_index} GREATER -1)
-  list(REMOVE_ITEM LLVMSupport_LIBS Z3)
-  list(APPEND LLVMSupport_LIBS z3::libz3)
-  set_target_properties(LLVMSupport PROPERTIES
-    INTERFACE_LINK_LIBRARIES "${LLVMSupport_LIBS}")
-endif()
+set(_VCPKG_SHARE_DIR "${VCPKG_ROOT}/installed/${VCPKG_TARGET_TRIPLET}/${CMAKE_INSTALL_DATADIR}")
 
-# google log module
-find_package(glog CONFIG QUIET)
-if (NOT glog_FOUND)
-  find_package(glog REQUIRED)
-  list(APPEND PROJECT_LIBRARIES glog)
-else()
-  list(APPEND PROJECT_LIBRARIES glog::glog)
-endif()
+find_package(gflags CONFIG REQUIRED HINTS "${_VCPKG_SHARE_DIR}")
+find_package(glog CONFIG REQUIRED HINTS "${_VCPKG_SHARE_DIR}")
+find_llvm(llvm "${_VCPKG_SHARE_DIR}")
```

Comment:
None of these hints should be required, and if they are, then the cxx-common vcpkg packaging should be changed.

What are the issues you are encountering here that makes the `_VCPKG_SHARE_DIR` required for use?

Reply:
I no longer remember, but it may have been remill.